python/xorbits/_mars/tensor/fuse/jax.py

```diff
@@ -34,61 +37,79 @@ class TensorJAXFuseChunk(TensorFuse, TensorFuseChunkMixin):
     @classmethod
     def execute(cls, ctx, op):
         chunk = op.outputs[0]
-        inputs = [ctx[c.key] for c in op.inputs]
+        inputs = [jnp.asarray(ctx[c.key]) for c in op.inputs]
         input_dict = {c.key: i for c, i in zip(op.inputs, inputs)}
-        res = np.array(_evaluate(chunk, input_dict))
+        res = np.asarray(_eval(input_dict, chunk))
         res = _maybe_keepdims(chunk, res)
         if chunk.ndim == 0 and res.ndim == 1 and res.size == 0:
             res = res.dtype.type(0)
         ctx[chunk.key] = res
 
 
-# recursively compute the inputs
-def _evaluate(chunk, input_dict):
-    op_type = type(chunk.op)
-
-    if chunk.key in input_dict:
-        return input_dict[chunk.key]
-    elif op_type is TensorJAXFuseChunk:
-        return _evaluate(chunk.composed[-1], input_dict)
-    elif op_type in ARITHMETIC_SUPPORT:
-        if hasattr(chunk.op, "inputs"):
-            if hasattr(chunk.op, "input"):
-                return _get_jax_function(chunk.op)(
-                    _evaluate(chunk.op.input, input_dict)
-                )
-            elif np.isscalar(chunk.op.lhs):
-                rhs = _evaluate(chunk.op.rhs, input_dict)
-                return _get_jax_function(chunk.op)(chunk.op.lhs, rhs)
-            elif np.isscalar(chunk.op.rhs):
-                lhs = _evaluate(chunk.op.lhs, input_dict)
-                return _get_jax_function(chunk.op)(lhs, chunk.op.rhs)
+if JAX_INSTALLED:
+
+    @ft.partial(jax.jit, static_argnums=(1,))
+    def _eval(input_dict, chunk):
```

Comment:
Almost every execution triggers jax jit recompilation. This may be optimized in the future.